src/transformers/generation_utils.py

```diff
@@ -905,6 +906,16 @@ def generate(
                 input_ids, pad_token_id, eos_token_id, inputs_embeds
             )
 
+        if self.config.is_encoder_decoder:
```

Comment:
can we remove this from the PR? This PR is for TF only no? I understand the need for this, but it's pretty hacky code that we should solve in a nicer way. I'll make the generate() function cleaner next week